### DISCUSSION TITLE
Is it possible to create Database WebHooks locally?

### DISCUSSION THREAD

**Discussion by JSmithOner** [supabase/supabase#9112] [Questions] [answered]
Everything is in the title.
I'm following the Happy Hour tutorial and they use database webhooks.
Only problem is that I develop using local docker image.
So is there any ways to create a local database webhook just like triggers and database functions?

Many thanks in advance,

**Comment by Ziinc** [comment, marked as answer]
Yes it is possible, just include them directly in your migrations and you will be able to trigger the functions through the `.rpc()` method on the client.
Triggers and webhooks just need the SQL to be executed on the database, it does not matter if it is hosted on docker or not.

**Reply by JSmithOner** [reply]
@Ziinc Thank you for your answer. Do you have any examples because I know how to create a trigger or a database function but I don't know how to create a webhook (can't find this in pg admin). Best.

**Reply by Ziinc** [reply]
Just to be clear, you are trying to receive the HTTP request? or execute a http request?

If receiving the http request, I would recommend using [edge functions](https://supabase.com/docs/guides/functions) for this as it is more scalable.

If it is executing the http request, you can use a postgres trigger together with the [pg_net extension](https://github.com/supabase/pg_net) to execute http requests

**Reply by JSmithOner** [reply]
@Ziinc I can't remember what I was trying to do at the moment I asked this question but I get it and your answer is pretty clear now.My point more globally about being sure I can reproduce my application on local environment without use of internet ( also I know it also depends on third services I'm using ) and the answer is that supabase cannot be used totally in a local environment as studio and serving functions is not the same a doing this online. Thank you again.Best

**Reply by Ziinc** [reply]
Yes, as of now, the local development integration for edge functions so as to be completely offline is not available.

**Reply by JSmithOner** [reply]
@Ziinc just an idea of enhancement out of the box. This would be nice if local supabase could be fully the same as the remote service. This would probably lower your costs and would be of great use when developing locally to perform the CI/CD pipeline you've showed in your documentation. Please do not feel no critics here only something that would be great to implement. Best.

**Reply by ikhsan** [reply]
> If it is executing the http request, you can use a postgres trigger together with the [pg_net extension](https://github.com/supabase/pg_net) to execute http requests

@Ziinc Do you mind sharing how to recreate the trigger that mimics the Database Webhook? 

[pg_net](https://github.com/supabase/pg_net/blob/master/docs/api.md#usage-1) allows firing the request, but I still don't know how to recreate the JSON payload similar to [the specification](https://supabase.com/blog/supabase-functions-updates#hook-payload). Maybe there is open-source code that we can follow?

**Comment by garysm** [comment]
Is it still the case that there is no integration between edge functions and webhooks locally? As mentioned before this would be a great feature to implement, especially for CI/CD/testing.
